string name
uint8 call_type